src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -636,7 +636,7 @@ private static boolean startsWithInheritDoc(DetailNode root) {
         boolean found = false;
         final DetailNode[] children = root.getChildren();
 
-        for (int i = 0; !found; i++) {
+        for (int i = 0; true; i++) {
```

Comment:
Replace this with a for-each loop.

Reply:
I am again facing 
![image](https://github.com/checkstyle/checkstyle/assets/97679350/21a8d5bd-8f91-4e09-a72f-8de7f86ef556)
 if i use for each loop

Reply:
@Kevin222004 Replace this with a while loop, the condition:
```java
child.getType() != JavadocTokenTypes.LEADING_ASTERISK && !CommonUtil.isBlank(child.getText())
``` 
would always be executed as every Javadoc tree has `EOF`.
https://github.com/checkstyle/checkstyle/blob/a174e6e47fdc1e6e11f7c5a9688a481a5274df6c/src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java#L1245

Reply:
@Kevin222004 , ping

Reply:
@Kevin222004 , can we use findfirst pattern
https://stackoverflow.com/a/65074853/1015848

Reply:
https://github.com/checkstyle/checkstyle/pull/13329#discussion_r1296652782 
The way how the logic is set in loop we can't use it